agent_build_refactored/utils/aws/ec2.py

```diff
@@ -98,15 +96,45 @@ def _create_ssh_container(
             toolset_image_name
         )
 
+        # The AWS_PRIVATE_KEY must be saved inside the running container using an echo command with a pipe.
+        # Using a volume mount is not possible when docker is installed via snap and leaves a key file in the filesystem.
         cmd_args.extend(
-            additional_cmd_args
+            [     
+                "/bin/bash",
+                "-c",
+                f"mkdir /tmp/mounts; echo \"{os.environ['AWS_PRIVATE_KEY']}\" > {self._ssh_client_container_in_docker_private_key_path}; chmod 600 {self._ssh_client_container_in_docker_private_key_path}; cat {self._ssh_client_container_in_docker_private_key_path}; " +
```

Comment:
Is it a good idea to cat private key file at the end? This could cause a secret leak in the logs, right?

I guess if the private key material (file content) is defined as a secret in Github Actions / similar, this should not be an issue, but still probably better to remove this step (I assume it was there only for troubleshooting / debugging or a similar reason).

Reply:
You're right! It's *-ed, but still a very bad practice. Will fix it.